File: compiler/GENERATOR/BYTECODE/TEST/BclassUser.java
/*
  Compilation statements: 
    (make clean; jbeta -x tstenv; \rm -f javabc/String.class)
    jbeta -x Bclass.bet
    javac -d javabc -classpath javabc BclassUser.java
    java -classpath javabc BclassUser
*/ 

class Jclass extends Bclass
{
  // Java constructor must transfer origin object to BETA constructor
  Jclass(tstenv origin)
    {
	super(origin);
    }

  public void compute(int arg)
    {
      System.out.println("Java: compute(" + arg + ") called. Calling BETA");
      super.compute(arg); 
    }
}


class BclassUser
{
    public static void main(String args[])
    {
	betaenv env = new betaenv(null);
	env.init();
	Bclass b = new Jclass(env);
	b.set(9);
	System.out.println(b.get());
	b.compute(10);
	System.out.println(b.get());
	b.compute(-6);
	System.out.println(b.get());
	b.compute(40);
	System.out.println(b.get());
    } 
    
}

File: compiler/GENERATOR/BYTECODE/TEST/BclassUser.cs
/*
  Compilation statements: 
    nbeta -x Bclass.bet
    csc /nologo /r:System.dll /r:dotnet/tstenv.dll /r:dotnet/Bclass.dll BclassUser.cs 
*/ 
using System;

class Cclass: Bclass
{
  // C# constructor must transfer origin object to BETA constructor
  public Cclass(tstenv origin):base(origin)
    {
    }

  public override void compute(int arg)
    {
      Console.WriteLine("C#:   compute(" + arg + ") called. Calling BETA");
      base.compute(arg); // super
    }
}

class BclassUser
{
  public static void Main() 
    {
      Bclass b = new Cclass(new tstenv(null));
      b.set(9);
      Console.WriteLine(b.get());
      b.compute(10);
      Console.WriteLine(b.get());
      b.compute(-6);
      Console.WriteLine(b.get());
      b.compute(40);
      Console.WriteLine(b.get());
    } 
}
